### DISCUSSION TITLE
ARKit 와 SceneKit 를 이용해 3D 모델(usdz파일)을 업로드 할 수 있는 테스트 코드입니다

### DISCUSSION THREAD

**Discussion by hyeonheebee** [DeveloperAcademy-POSTECH/.github#30] [Show and tell]
NC2에서 오후 에이치와 ARKit 를 공부를 진행하면서 ARKit 와 SceneKit 를 이용해 3D 모델(usdz파일)을 업로드 하고 핸드제스쳐(탭, 핀치, 펜, 롱프레스 제스쳐 등)를 활용하여 3D 모델 객체를 변형, 애니메이션을 적용하는 프로젝트를 진행했습니다. 

저희 프로젝트를 아주아주 간소화하여 가벼운 코드로 만들어보았어요. 
ARKit 에 대한 아주 기초적인 이해와 3D 오브젝트를 올리고 애니메이션을 적용하거나 핸드제스쳐로 객체를 변형시켜 볼 수 있는 코드에요. 
처음 ARKit 와 SceneKit을 적용하실때도 이해하기 쉬우시도록 주석을 달아놓았어요. 

**_단순히 제가 이해한 선에서 도움이 되었으면 하는 마음으로 테스트 코드를 만들어 보았어요. 
부족한 부분이 많으니 함께 유지보수 하고 댓글로도 많은 피드백 부탁드려요 :)_** 

## 코드에 대한 설명 
현재 usdz 모델은 candle 하나가 올라가있고 빌드하시게 되면 촛불이 아이폰 화면 AR 가상화면에 보이실거에요. 
해당 프로젝트에서 할 수 있는 (코드로 적용되어있는) 제스쳐는 다음과 같아요. 

1. 탭 제스쳐 : 해당 캔들의 불꽃노드를 탭해서 촛불을 껐다 킬 수 있게 되어있어요 
2. 핀치 제스쳐 : 엄지 검지를 통해 촛불을 확대, 축소할 수 있어요 
3. 펜 제스쳐 : 드래그를 통해 촛불의 위치를 이동할 수 있어요 

추가로 여러분이 활용할 수 있는 애니메이션은 두가지가 들어 있어요 (회전, 업다운) 

기타 사항은 댓글로 문의주시거나 팀즈로 문의주시고 주석도 함께 참고해주시면 감사하겠습니다 :) 

## 코드 깃허브 링크 

https://github.com/hyeonheebee/ARKit_Test

**Comment by yeniful** [comment]
우와 따끈따끈 새 글이군요 지식과 자료 공유해주셔서 감사드려요 🙌

**Comment by manju-minji** [comment]
우와 멋진 공유 감사드려요!!
